spec/requests/api/v1/customer_request_spec.rb

```diff
@@ -0,0 +1,138 @@
+require 'rails_helper'
+
+describe "customer requests" do 
+  it "returns data for all customers" do 
+    create_list(:customer,3)
+
+    get "/api/v1/customers"
+    customers = JSON.parse(response.body)
+
+    expect(response).to be_successful 
+    expect(customers.first).to have_key("first_name")
+  end
+  it "can find a single  customer by id" do 
+    create(:customer, id: 1)
+
+    get "/api/v1/customers/1"
+    customer = JSON.parse(response.body)
+
+    expect(response).to be_successful
+    expect(customer["id"]).to eq(1)
+  end
+  it "can find a customer from a given id" do 
+    create(:customer, id: 2)
+
+    get "/api/v1/customers/find?id=2"
+    customer = JSON.parse(response.body)
+
+    expect(response).to be_successful
+    expect(customer["id"]).to eq(2)
+  end
+
+  it "can find all the customers with a given id" do 
+     create(:customer, id: 2)
+     create(:customer, id: 3)
+
+      get "/api/v1/customers/find_all?id=2"
+
+      customer = JSON.parse(response.body)
+      
+      expect(response).to be_successful
+      expect(customer.first["id"]).to eq(2)
+  end
+
+  it "can find a customer given a first name" do 
+    create(:customer, first_name: "Nico")
+
+    get "/api/v1/customers/find?first_name=Nico"
+
+    customer = JSON.parse(response.body)
+
+    expect(response).to be_successful
+    expect(customer["first_name"]).to eq("Nico")
+  end
+
+  it "can find all customers given a first name" do 
+    create(:customer, first_name: "Nico")
+    create(:customer, first_name: "Nico")
+
+    get "/api/v1/customers/find_all?first_name=Nico"
+
```

Comment:
Make spacing consistent here.

Reply:
There are a few more like this further down: try to be consistent with setup vs. expectations/assertions if that's the route you're wanting to go.